### PR TITLE
starlark: dedent continued function param lines

### PR DESCRIPTION
**Pull request by alexeagle**
This allows multi-line parameter documentation under Args: without extra indents.
Fixes https://github.com/bazelbuild/stardoc/issues/74

### CODE REVIEW

**Comment by alexeagle** [comment]
ping @c-parsons are you the right person to merge?

File: src/tools/starlark/java/com/google/devtools/starlark/common/DocstringUtils.java [hunk 1]
```diff
@@ -479,15 +479,21 @@ private String checkForNonStandardDeprecation(String line) {
...
           description.append('\n');
           continue;
         }
-        if (getIndentation(line) <= baselineIndentation) {
+        int lineIndentation = getIndentation(line);
+        if (lineIndentation <= baselineIndentation) {
           break;
         }
-        String trimmedLine = line.substring(baselineIndentation);
+        if (lineIndentation < continuationIndentation) {
```
Comment (c-parsons) [inline src/tools/starlark/java/com/google/devtools/starlark/common/DocstringUtils.java:493]: We're not truly finding the minimum indentation value here, as we're still processing as we go.
Consider the difference in behavior for:

```
  Line one
          Line two
                    Line three
```

versus

```
               Line one
        Line two
   Line three
```

**Review by c-parsons — COMMENTED** [review]
Could you modify `pure_markdown_test` as well to demonstrate how this change affects markdown tables?

**Comment by c-parsons** [comment]
FYI @alandonovan @brandjon

File: src/tools/starlark/java/com/google/devtools/starlark/common/DocstringUtils.java (hunk 1 above)
Comment (alexeagle) [reply src/tools/starlark/java/com/google/devtools/starlark/common/DocstringUtils.java:493]: correct, but the failure mode is that you get some extra whitespace if the continuation is accidentally not consistent. For your example I think

```
Line one
        Line two
                  Line three
```
and
```
Line one
Line two
Line three
```
are the best outcome we'll get without scanning ahead for the minimal intent. Do you think it's worth adding the complexity to scan ahead?

**Comment by alexeagle** [comment]
Added the change to `pure_markdown_template_test` as requested :)

File: src/tools/starlark/java/com/google/devtools/starlark/common/DocstringUtils.java (hunk 1 above)
Comment (c-parsons) [reply src/tools/starlark/java/com/google/devtools/starlark/common/DocstringUtils.java:493]: Yes, I don't think it's much more complex to scan ahead (we're not dealing with gigantic strings here, so it should be fine to store all the lines in a list, take the min indent, and then iterate over)

I foresee later user confusion / frustration with the current behavior, if we don't fix this right the first time.

Thanks!

**Comment by c-parsons** [comment]
FYI @alandonovan

File: src/tools/starlark/java/com/google/devtools/starlark/common/DocstringUtils.java (hunk 1 above)
Comment (alexeagle) [reply src/tools/starlark/java/com/google/devtools/starlark/common/DocstringUtils.java]: Okay, done PTAL

**Comment by alexeagle** [comment]
ping @c-parsons mergey merge-merge?

**Comment by c-parsons** [comment]
Merge in review. Thanks for your patience.

**Pull request closed without merging**